azerothcore/azerothcore-wotlk
# fix(Core): Pet being slowed when owner is slowed

### fix(Core): Pet being slowed when owner is slowed

## Changes Proposed:
- FollowMovementGenerator updates the Pets Movement speed to match their owners, lets add a check to see if the owner currently suffers from an Aura that slows him down, if yes then don't slow down the pet aswell.


## Issues Addressed:
- Closes: https://github.com/azerothcore/azerothcore-wotlk/issues/4975
- Closes: https://github.com/azerothcore/azerothcore-wotlk/issues/4772
- Closes: https://github.com/chromiecraft/chromiecraft/issues/279



## SOURCE:



## Tests Performed:

- Create Hunter/Warlock
- Summon Pet
- Get dazed or otherwise slowed by a spell
- Pet won't be slowed anymore


## How to Test the Changes:

- Apply patch.
- Create Hunter/Warlock.
- Get into a fight so a creature hits you until you are dazed.
- Let your pet attack another creature.
- Your pet should not be slowed anymore.

**Edit by Shin**: also test pets in general and make sure there is no regression

## Known Issues and TODO List:

- There seem to be certain edge cases where "owner->GetOwner()->GetAuraEffectsByType(SPELL_AURA_MOD_DECREASE_SPEED).empty()" returns true even when you have a slowing effect on you. This seems to happen to me when you had a certain combination of slow down effects once (even if they aren't active anymore).
- I have no idea why that happens and it seems to be a problem with the way we handle our lists, this problem persists until you restart the server. Is there a better way to check if the player is slowed?


## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
